dotnet-engine/Yggdrasil.Engine/FFI.cs

```diff
@@ -99,4 +99,16 @@ public static IntPtr ListKnownToggles(IntPtr ptr)
     {
         return list_known_toggles(ptr);
     }
+
+    /// <summary>
+    /// Converts a string to a UTF-8 encoded null-terminated byte array.
+    /// </summary>
+    /// <param name="input">The string to convert.</param>
+    /// <returns>A UTF-8 encoded null-terminated byte array.</returns>
+    private static byte[] ToUtf8Bytes(string input)
+    {
+        byte[] utf8Bytes = System.Text.Encoding.UTF8.GetBytes(input);
+        Array.Resize(ref utf8Bytes, utf8Bytes.Length + 1);
```

Comment:
This appears to make space for the null terminator but never gets added. Do you know what .NET does if you don't explicitly set the null byte? Can we rather just set it? It's probably fine but I see the Rust docs say that getting this wrong means UB and I'm trying to avoid that like the plague

Reply:
Nevermind. I read the docs. This should be fine

Reply:
For more context: This should expand the existing array and append a 0 (default) byte automatically, which is what we want for our null terminator.